.github/workflows/_run_test.yml

```diff
@@ -0,0 +1,102 @@
+        type: string
+        description: Runner to use for test
+        required: true
+      TIMEOUT:
+        type: number
+        description: Max runtime of test in minutes
+        required: false
+        default: 10
+      SCRIPT:
+        type: string
+        description: Test script to execute
+        required: true
+      AFTER_SCRIPT:
+        type: string
+        description: Script to run after main test
+        required: false
+        default: ":"
+      IS_OPTIONAL:
+        type: boolean
+        description: Failure will cancel all other tests if set to true
+        required: false
+        default: false
+    outputs:
+      conclusion:
+        description: Conclusion of main test step
+        value: ${{ jobs.main.outputs.conclusion }}
+      log:
+        description: Last 2000 characters of the test step's log
+        value: ${{ jobs.main.outputs.log }} 
+jobs:
+  
+  main:
+    runs-on: ${{ inputs.RUNNER }} 
+    outputs:
+      conclusion: ${{ steps.main.conclusion }}
+      log: ${{ steps.main.outputs.log }}
+    steps:
+        - name: Docker system cleanup
+          run: |
+            docker system prune -a --filter "until=48h" --force || true
+
+        - name: Docker pull image
+          run: |
+            #docker pull nemoci.azurecr.io/nemo_aligner_container:${{ github.run_id }}
+            docker pull nemoci.azurecr.io/nemo_aligner_container:${{ github.event.pull_request.number }}
+
+        - id: main
+          name: Run main script
+          timeout-minutes: ${{ inputs.TIMEOUT }}
+          run: |
+            nvidia-smi
+            mkdir -p ${{ github.run_id }}
+            cd ${{ github.run_id }}/
+            set +e
+            (
+              set -e
+
+              cmd=$(cat <<"RUN_TEST_EOF"
+            # Sanity check the driver/cuda combo
+            cudaCheck
+            # In case git commands need to be run inside Aligner
+            git config --global --add safe.directory $ALIGNER_REPO_DIR
+            ${{ inputs.SCRIPT }}
+            RUN_TEST_EOF
+            )
+              #docker run --rm --runtime=nvidia --gpus all --shm-size=8g --env TRANSFORMERS_OFFLINE=0 --env HYDRA_FULL_ERROR=1 --volume /mnt/datadrive/TestData:/home/TestData:ro nemoci.azurecr.io/nemo_aligner_container:${{ github.run_id }} bash -eux -o pipefail -c "$cmd"
+              docker run --rm --runtime=nvidia --gpus all --shm-size=8g \
+                --env TRANSFORMERS_OFFLINE=0 --env HYDRA_FULL_ERROR=1 --env HF_HOME=/home/TestData/aligner/hf_home --env ALIGNER_CI_DIR=/home/TestData/aligner --env ALIGNER_REPO_DIR=/opt/NeMo-Aligner \
+                --volume /mnt/datadrive/TestData/aligner/checkpoints:/home/TestData/aligner/checkpoints:ro \
+                --volume /mnt/datadrive/TestData/aligner/hf_home/hub:/home/TestData/aligner/hf_home/hub:ro \
+                nemoci.azurecr.io/nemo_aligner_container:${{ github.event.pull_request.number }} \
+                bash -eux -o pipefail -c "$cmd"
```

Comment:
Dont need to change it in this PR, but it's unsafe to use `docker run` with GitHub actions & our infra. See other discussion

Reply:
addressed in https://github.com/NVIDIA/NeMo-Aligner/pull/356